haskell/private/packages.bzl

```diff
@@ -0,0 +1,47 @@
+"""Package list handling"""
+
+load(":private/set.bzl", "set")
+
+def expose_packages(build_info, lib_info, use_direct, use_my_pkg_id):
+    """
+    Returns the list of command line argument which should be passed
+    to GHC in order to enable haskell packages.
+    """
+
+    args = [
+        # In compile.bzl, we pass this just before all -package-id
+        # arguments. Not doing so leads to bizarre compile-time failures.
+        # It turns out that equally, not doing so leads to bizarre
+        # link-time failures. See
+        # https://github.com/tweag/rules_haskell/issues/395.
+        "-hide-all-packages",
+    ]
+
+    # Expose all prebuilt dependencies
+
+    # We have to remember to specify all (transitive) wired-in
+    # dependencies or we can't find objects for linking
+
+    # XXX This should be really dep_info.direct_prebuilt_deps, but since we
```

Comment:
The pattern elsewhere with the comment tokens is `XXX:`.

Reply:
Good catch, will be corrected on nix iteration.